modules/services/osmscout-server.nix

```
@@ -0,0 +1,89 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.services.osmscout-server;
+in {
+  meta.maintainers = [ maintainers.Thra11 ];
+
+  options = {
+    services.osmscout-server = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable OSM Scout Server as a systemd service.
+        '';
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.osmscout-server;
+        defaultText = "pkgs.osmscout-server";
+        description = ''
+          The OSM Scout Server package to run.
+        '';
+      };
+
+      network = {
+        startWhenNeeded = mkOption {
+          type = types.bool;
+          default = true;
+          description = ''
+            Enable systemd socket activation.
+          '';
+        };
+
+        listenAddress = mkOption {
+          type = types.str;
+          default = "127.0.0.1";
+          description = ''
+            The address for the server to listen on.
+          '';
+        };
+
+        port = mkOption {
+          type = types.port;
+          default = 8553;
+          description = ''
+            The TCP port on which the server will listen.
+          '';
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      (lib.hm.assertions.assertPlatform "services.osmscout-server" pkgs
+        lib.platforms.linux)
+    ];
+
+    systemd.user.services.osmscout-server = {
+      Unit = { Description = "OSM Scout Server"; };
+
+      Install = mkIf (!cfg.network.startWhenNeeded) {
+        WantedBy = [ "default.target" ];
+      };
+
+      Service = {
+        ExecStart = "${cfg.package}/bin/osmscout-server --systemd --quiet";
+      };
+    };
+
+    systemd.user.sockets.osmscout-server = mkIf cfg.network.startWhenNeeded {
+      Unit = { Description = "OSM Scout Server Socket"; };
+
+      Socket = {
+        ListenStream =
+          "${cfg.network.listenAddress}:${toString cfg.network.port}";
+        TriggerLimitIntervalSec = "60s";
+        TriggerLimitBurst = 1;
+      };
+
+      Install = { WantedBy = [ "user-session.target" ]; };
```

`user-session.target` is used in the systemd target generated by upstream osmscout-server, but I'm not sure if it's the right target in this context: It doesn't appear to exist by default.

---

No, there is no such target inside a user session. You'll want
```suggestion
      Install = { WantedBy = [ "sockets.target" ]; };
```
instead.